### PR TITLE
Ros2 development

### PR DESCRIPTION
**Pull request by rafal-gorecki**

### CODE REVIEW

File: Dockerfile.desktop [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    python3-flake8-docstrings \
+    python3-pip \
+    python3-pytest-cov \
+    ros-dev-tools && \
+    pip3 install \
+    argcomplete \
+    numpy \
+    empy \
+    lark && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (KmakD) [inline Dockerfile.desktop]:
```suggestion
RUN apt-get update || true && \
    apt-get install --no-install-recommends -y \
        ros-${ROS_DISTRO}-ros-base \
        ros-${ROS_DISTRO}-ament-cmake-clang-format \
        ros-${ROS_DISTRO}-image-transport \
        ros-${ROS_DISTRO}-image-transport-plugins \
        ros-${ROS_DISTRO}-diagnostic-updater \
        ros-${ROS_DISTRO}-xacro \
        python3-flake8-docstrings \
        python3-pip \
        python3-pytest-cov \
        ros-dev-tools && \
    pip3 install \
        argcomplete \
        numpy \
        empy \
        lark && \
    rm -rf /var/lib/apt/lists/*
```

File: Dockerfile.desktop [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
```
Comment (KmakD) [inline Dockerfile.desktop]:
```suggestion
RUN apt-get update || true && \
    apt-get install --no-install-recommends -y \
        ros-${ROS_DISTRO}-ros-base \
        ros-${ROS_DISTRO}-ament-cmake-clang-format \
        ros-${ROS_DISTRO}-image-transport \
        ros-${ROS_DISTRO}-image-transport-plugins \
        ros-${ROS_DISTRO}-diagnostic-updater \
        ros-${ROS_DISTRO}-xacro \
        python3-flake8-docstrings \
        python3-pip \
        python3-pytest-cov \
        ros-dev-tools && \
    pip3 install \
        argcomplete \
        numpy \
        empy \
        lark && \
    rm -rf /var/lib/apt/lists/*
```

File: Dockerfile.jetson [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    python3-flake8-docstrings \
+    python3-pip \
+    python3-pytest-cov \
+    ros-dev-tools && \
+    pip3 install \
+    argcomplete \
+    numpy \
+    empy \
+    lark && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (KmakD) [inline Dockerfile.jetson]:
```suggestion
RUN apt-get update || true && \
    apt-get install --no-install-recommends -y \
        ros-${ROS_DISTRO}-ros-base \
        ros-${ROS_DISTRO}-ament-cmake-clang-format \
        ros-${ROS_DISTRO}-image-transport \
        ros-${ROS_DISTRO}-image-transport-plugins \
        ros-${ROS_DISTRO}-diagnostic-updater \
        ros-${ROS_DISTRO}-xacro \
        build-essential \
        git \
        python3-colcon-common-extensions \
        python3-colcon-mixin \
        python3-rosdep \
        python3-vcstool \
        python3-flake8-docstrings \
        python3-pip \
        python3-pytest-cov \
        ros-dev-tools && \
    pip3 install \
        argcomplete \
        numpy \
        empy \
        lark && \
    rm -rf /var/lib/apt/lists/*
```

File: Dockerfile.jetson [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
```
Comment (KmakD) [inline Dockerfile.jetson]:
```suggestion
RUN apt-get update || true && \
    apt-get install --no-install-recommends -y \
        ros-${ROS_DISTRO}-ros-base \
        ros-${ROS_DISTRO}-ament-cmake-clang-format \
        ros-${ROS_DISTRO}-image-transport \
        ros-${ROS_DISTRO}-image-transport-plugins \
        ros-${ROS_DISTRO}-diagnostic-updater \
        ros-${ROS_DISTRO}-xacro \
        build-essential \
        git \
        python3-colcon-common-extensions \
        python3-colcon-mixin \
        python3-rosdep \
        python3-vcstool \
        python3-flake8-docstrings \
        python3-pip \
        python3-pytest-cov \
        ros-dev-tools && \
    pip3 install \
        argcomplete \
        numpy \
        empy \
        lark && \
    rm -rf /var/lib/apt/lists/*
```

File: Dockerfile.jetson [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    source /opt/ros/$ROS_DISTRO/setup.bash && \
+    rosdep init && \
+    rosdep update --rosdistro $ROS_DISTRO && \
+    rosdep install --from-paths src -y -i && \
+    rm -rf /var/lib/apt/lists/*
+
+# Setup environment variables
+COPY ./ros_entrypoint.sh /ros_entrypoint.sh
+
+ENTRYPOINT ["/ros_entrypoint.sh"]
```
Comment (KmakD) [inline Dockerfile.jetson]:
```suggestion
COPY ./ros_entrypoint.sh /ros_entrypoint.sh
ENTRYPOINT ["/ros_entrypoint.sh"]
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,63 @@
...
+
+**ROS1:**
+- **`zed-desktop:noetic`** for desktop platform with CUDA (tested on platform with 11.7).
+- **`zed-jetson:noetic`** for Jetson platform currently support - **Jetson Xavier, Orin AGX/NX/Nano, CUDA 11.4** (tested on Xavier AGX).
+- **`husarion/zed-jetson:melodic`** for Jetson platform currently not supported - **Jetson Nano, TX2/TX2 NX, CUDA 10.2** (tested on Nano).
+
+**ROS2:**
+- **`husarion/zed-desktop:humble`** for desktop platform with CUDA (tested on platform with 11.7).
+- **`husarion/zed-jetson:humble`** for Jetson platform currently support - **Jetson Xavier, Orin AGX/NX/Nano, CUDA 11.4** (tested on Xavier AGX).
+- **`no image`** for Jetson platform currently not supported - **Jetson Nano, TX2/TX2 NX, CUDA 10.2** (tested on Nano).
```
Comment (KmakD) [inline README.md]:
```suggestion
```
I don't see a point in writing info about non existing image

File: README.md [hunk 7]
```diff
@@ -1,2 +1,63 @@
...
+## Running demo with Docker compose
+
+Connect ZED camera to your platform, `export CAMERA_MODEL` and run following commands:
+
+```bash
+cd demo
+export DISPLAY=:1
+xhost local:root
+docker compose up
+```
```
Comment (KmakD) [inline README.md:61]: I would move that before `development` section

File: demo/compose.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
+services:
+        
+  zed:
+    image: ${ZED_IMAGE}
+    container_name: zed
+    network_mode: host
+    runtime: nvidia
```
Comment (KmakD) [inline demo/compose.yaml:7]: Don't we need:
```
environment:
      - DISPLAY
      - QT_X11_NO_MITSHM=1
      - NVIDIA_VISIBLE_DEVICES=all
      - NVIDIA_DRIVER_CAPABILITIES=compute,utility,display
```
for nvidia to work properly?

File: demo/compose.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    tty: true
+    volumes:
+      - ./params:/ros_ws/src/zed-ros-wrapper/zed_wrapper/params
+      - /dev:/dev
```
Comment (KmakD) [inline demo/compose.yaml]:
```suggestion
```
As we are using privileged this should not be necessary

File: demo/compose.yaml [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    container_name: zed
+    network_mode: host
+    runtime: nvidia
+    tty: true
+    volumes:
+      - ./params:/ros_ws/src/zed-ros-wrapper/zed_wrapper/params
+      - /dev:/dev 
+      - /tmp/zed_ai/:/usr/local/zed/resources/
+    privileged: true
+    command: ros2 launch zed_wrapper zed2.launch.py
```
Comment (KmakD) [inline demo/compose.yaml]:
```suggestion
    command: ros2 launch zed_wrapper ${CAMERA_MODEL}.launch.py
```
I guess launch file should be env

**Review by KmakD — CHANGES_REQUESTED** [review]

File: demo/compose.yaml (hunk 9 above)
Comment (rafal-gorecki) [reply demo/compose.yaml]: I guess it is for replug / hot plug

**Pull request merged**